pootle/apps/virtualfolder/models.py

```diff
@@ -347,19 +275,156 @@ def get_adjusted_pootle_path(self, pootle_path):
         trail = pootle_path.replace(lead, '').lstrip('/')
         return '/'.join([lead, self.name, trail])
 
-    def get_translate_url(self, pootle_path, **kwargs):
-        """Get the translate URL for this virtual folder in the given path."""
-        adjusted_path = self.get_adjusted_pootle_path(pootle_path)
-        lang, proj, dp, fn = split_pootle_path(adjusted_path)
 
+class VFolderCachedTreeItemManager(models.Manager):
+    use_for_related_fields = True
+
+    def get_queryset(self):
+        return super(VFolderCachedTreeItemManager, self) \
+            .get_queryset().select_related('vfolder')
+
+
+class VFolderCachedTreeItem(models.Model, CachedTreeItem):
```

Comment:
As I was discussing with @ta2-1 the other day, we are very confused to have a model named after *CachedTreeItem*, giving the false impression this object comes from Redis directly, when it's a DB model.

We are not either sure of the need for a separate model here (there's also `VirtualFolder`, which isn't cached).